diff --git a/drivers/rtc/Makefile b/drivers/rtc/Makefile
index bfb574641..6a65ee62e 100644
--- a/drivers/rtc/Makefile
+++ b/drivers/rtc/Makefile
@@ -125,7 +125,7 @@ obj-$(CONFIG_RTC_DRV_PCF8563)	+= rtc-pcf8563.o
 obj-$(CONFIG_RTC_DRV_PCF8583)	+= rtc-pcf8583.o
 obj-$(CONFIG_RTC_DRV_PIC32)	+= rtc-pic32.o
 obj-$(CONFIG_RTC_DRV_PL030)	+= rtc-pl030.o
-obj-$(CONFIG_RTC_DRV_PL031)	+= rtc-pl031.o
+#obj-$(CONFIG_RTC_DRV_PL031)	+= rtc-pl031.o
 obj-$(CONFIG_RTC_DRV_PM8XXX)	+= rtc-pm8xxx.o
 obj-$(CONFIG_RTC_DRV_PS3)	+= rtc-ps3.o
 obj-$(CONFIG_RTC_DRV_PXA)	+= rtc-pxa.o
diff --git a/include/linux/interrupt.h b/include/linux/interrupt.h
index ee8299eb1..8821299da 100644
--- a/include/linux/interrupt.h
+++ b/include/linux/interrupt.h
@@ -538,6 +538,8 @@ enum
 	HRTIMER_SOFTIRQ,
 	RCU_SOFTIRQ,    /* Preferable RCU should always be the last softirq */
 
+    HELLO_SOFTIRQ,  /* add by wit for softirq test */
+
 	NR_SOFTIRQS
 };
 
diff --git a/kernel/softirq.c b/kernel/softirq.c
index 09229ad82..a4ae5d4cd 100644
--- a/kernel/softirq.c
+++ b/kernel/softirq.c
@@ -479,6 +479,8 @@ void raise_softirq(unsigned int nr)
 	local_irq_restore(flags);
 }
 
+EXPORT_SYMBOL(raise_softirq);
+
 void __raise_softirq_irqoff(unsigned int nr)
 {
 	lockdep_assert_irqs_disabled();
@@ -491,6 +493,8 @@ void open_softirq(int nr, void (*action)(struct softirq_action *))
 	softirq_vec[nr].action = action;
 }
 
+EXPORT_SYMBOL(open_softirq);
+
 /*
  * Tasklets
  */
